modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -514,11 +566,25 @@ def map_func(left, right=right, kwargs=kwargs):
     def reindex(self, axis, labels, **kwargs):
         new_index = self.index if axis else labels
         new_columns = labels if axis else self.columns
+        actual_index = None
+        if "_reset_index" in kwargs:
```

Comment:
how does this happen?

Reply:
We have to force the index of the right operand to be overwritten by the index of the left operand in order for the dataframe to be correct after the `concat` operation.